### ISSUE TITLE
fn with name is marked as unused

### ISSUE THREAD

**Issue by Frozenlock** [clojure-lsp/clojure-lsp#250]
I think this is a bug.

In the following snippet, `my-fn` is marked as unused:
```clj
(let [my-fn2 (fn my-fn [n]
               (when-not (<= n 0)
                 (my-fn (dec n))))]
  (my-fn2 10))
```
![image](https://user-images.githubusercontent.com/1152041/104244850-3ee11080-5431-11eb-80d0-f2b49a71a95c.png)

**Comment by ericdallo** [comment]
Thanks for the report!
Yeah, It's a bug indeed, I think this case was not handled very well

**Comment by ericdallo** [comment]
Hey @Frozenlock I tested and this was fixed after #261 :) LMK if any issues
